lib/auth/auth.go

```diff
@@ -5371,6 +5380,79 @@ func (a *Server) syncDynamicLabelsAlert(ctx context.Context) {
 	}
 }
 
+func (a *Server) cleanupNotifications(ctx context.Context) {
+	userNotifications, _, err := a.Cache.ListUserNotifications(ctx, 0, "")
+	if err != nil {
+		slog.WarnContext(ctx, "failed to list first set of notifications for periodic cleanup", err)
+	}
+	globalNotifications, _, err := a.Cache.ListGlobalNotifications(ctx, 0, "")
+	if err != nil {
+		slog.WarnContext(ctx, "failed to list second set of notifications for periodic cleanup", err)
+	}
+
+	timeNow := time.Now()
+
+	// Convert into a map where the key is the notification id.
+	globalNotificationsIdMap := make(map[string]*notificationsv1.GlobalNotification)
+	for _, gn := range globalNotifications {
+		globalNotificationsIdMap[gn.GetMetadata().GetName()] = gn
+	}
+
+	// Delete expired global notifications.
+	for notificationId, gn := range globalNotificationsIdMap {
+		expiry := gn.GetSpec().GetNotification().GetMetadata().GetExpires()
+
+		if timeNow.After(expiry.AsTime()) {
+			if err := a.DeleteGlobalNotification(ctx, notificationId); err != nil {
+				slog.WarnContext(ctx, "encountered error attempting to cleanup notification", err)
+			}
+			// Also remove it from the map so that we can pass it to the notification states cleanup
+			// without needing to list notifications again.
+			delete(globalNotificationsIdMap, notificationId)
+		}
+	}
+
+	// Convert into a map where the key is the notification id.
+	userNotificationsIdMap := make(map[string]*notificationsv1.Notification)
+	for _, un := range userNotifications {
+		userNotificationsIdMap[un.GetMetadata().GetName()] = un
+	}
+
+	// Delete expired user notifications.
+	for notificationId, gn := range userNotificationsIdMap {
```

Comment:
```suggestion
	for notificationID, un := range userNotificationsIdMap {
```

Remember to capitalize your initialisms.

And it looks like a copy paste leftover, but lets not use a variable name of `gn` for a user notification.